checker/jdk/signedness/src/java/lang/Integer.java

```diff
@@ -0,0 +1,1602 @@
+/*
+ * Copyright (c) 1994, 2013, Oracle and/or its affiliates. All rights reserved.
+ * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ */
+
+package java.lang;
+
+import java.lang.annotation.Native;
+
+import org.checkerframework.checker.index.qual.NonNegative;
+import org.checkerframework.checker.index.qual.Positive;
+import org.checkerframework.checker.signedness.qual.PolySigned;
+import org.checkerframework.checker.signedness.qual.Unsigned;
+import org.checkerframework.checker.signedness.qual.UnknownSignedness;
+import org.checkerframework.checker.signedness.qual.SignednessGlb;
+
+/**
+ * The {@code Integer} class wraps a value of the primitive type
+ * {@code int} in an object. An object of type {@code Integer}
+ * contains a single field whose type is {@code int}.
+ *
+ * <p>In addition, this class provides several methods for converting
+ * an {@code int} to a {@code String} and a {@code String} to an
+ * {@code int}, as well as other constants and methods useful when
+ * dealing with an {@code int}.
+ *
+ * <p>Implementation note: The implementations of the "bit twiddling"
+ * methods (such as {@link #highestOneBit(int) highestOneBit} and
+ * {@link #numberOfTrailingZeros(int) numberOfTrailingZeros}) are
+ * based on material from Henry S. Warren, Jr.'s <i>Hacker's
+ * Delight</i>, (Addison Wesley, 2002).
+ *
+ * @author  Lee Boynton
+ * @author  Arthur van Hoff
+ * @author  Josh Bloch
+ * @author  Joseph D. Darcy
+ * @since JDK1.0
+ */
+public final class Integer extends Number implements Comparable<Integer> {
+    /**
+     * A constant holding the minimum value an {@code int} can
+     * have, -2<sup>31</sup>.
+     */
+    @Native public static final int   MIN_VALUE = 0x80000000;
+
+    /**
+     * A constant holding the maximum value an {@code int} can
+     * have, 2<sup>31</sup>-1.
+     */
+    @Native public static final @Positive int   MAX_VALUE = 0x7fffffff;
```

Comment:
Throughout, I suggest using `@SignedPositive` rather than `@Positive` and `@NonNegative`.
Using `@Positive` makes it look like the code is annotated for the Index Checker, which it is not.

Reply:
It may give error because earlier when these fields were not annotated then error of `conflicting annotations` were occurring then I annotated them like this.
Check [this](https://groups.google.com/d/msg/checker-framework-gsoc/qeDMhtFW_uc/AxFyOaXtCgAJ)

Reply:
Exactly what error may it give?
Is there a bug report in the issue tracker?  If not, please add one.  If there is, you could point me at it.

Reply:
> Exactly what error may it give?

It will give the error `type.invalid.conflicting.annos` when files like LessThanValue.java and some guava files will get compiled.
For more details check this travis [link](https://travis-ci.org/typetools/checker-framework/builds/561164283?utm_source=github_status&utm_medium=notification).
> Is there a bug report in the issue tracker? If not, please add one. If there is, you could point me at it.

Is this a bug? It is looking to me more like a property. The thing which is happening is that when the JDK gets compiled it gets two different annotations (One from annotated JDK of signedness checker and one from Index Checker), which are getting conflicted.

Reply:
I think you are saying that the user can write two reasonable annotations that are not in conflict with one another, and the Checker Framework issues an undesirable when both are present.  That is a bug; could you please report it on the issue tracker?

(But, I'm not sure because you did not give a concrete example.)

Reply:
@gokaco Have you addressed this comment?

Reply:
> I think you are saying that the user can write two reasonable annotations that are not in conflict with one another, and the Checker Framework issues an undesirable when both are present. That is a bug; could you please report it on the issue tracker?

I have issued an issue related to this in the issue tracker. Please check it! #2641